Percona operator v2.8.x crashes with nil pointer in EnvFromSecrets

## Description

When using `percona.enabled: true` in the DittoServer CR, the Percona PostgreSQL Operator (both v2.8.0 and v2.8.2) crashes with a nil pointer dereference in `EnvFromSecrets` at `perconapgcluster_types.go:1114`.

## Root Cause

The PerconaPGCluster spec created by our operator includes `secrets: {}` (the Go zero value for the Secrets struct). The Percona operator's cache indexer calls `EnvFromSecrets()` which doesn't handle nil pointer fields within the `Secrets` struct, causing a panic.

## Stack Trace

```
panic: runtime error: invalid memory address or nil pointer dereference

github.com/percona/percona-postgresql-operator/v2/pkg/apis/pgv2.percona.com/v2.(*PerconaPGCluster).EnvFromSecrets
    perconapgcluster_types.go:1114
github.com/percona/percona-postgresql-operator/v2/pkg/apis/pgv2.percona.com/v2.init.func2
    perconapgcluster_types.go:1141
sigs.k8s.io/controller-runtime/pkg/cache.indexByField.func1
    informer_cache.go:236
```

## PerconaPGCluster Spec (minimal repro)

```yaml
apiVersion: pgv2.percona.com/v2
kind: PerconaPGCluster
metadata:
  name: test-postgres
spec:
  crVersion: 2.8.0
  postgresVersion: 16
  instances:
    - name: instance1
      replicas: 1
      dataVolumeClaimSpec:
        accessModes: [ReadWriteOnce]
        resources:
          requests:
            storage: 10Gi
  users:
    - name: dittofs
      databases: [dittofs]
  secrets: {}        # <-- This causes the crash
  backups:
    enabled: false
```

## Workaround

Manually deploy PostgreSQL (Deployment + Service) and use `database.postgresSecretRef` instead of `percona.enabled`.

## Fix Options

1. **In our operator**: Populate the `Secrets` field with non-nil sub-fields, or use `omitempty` to avoid emitting `secrets: {}` 
2. **Upstream fix**: Report to Percona - `EnvFromSecrets` should handle nil pointers gracefully

## Environment

- Kubernetes v1.35.0 (Scaleway Kapsule)
- Percona Operator v2.8.0 and v2.8.2 (both affected)
- DittoFS Operator v0.7.0

## Labels

- bug
- operator
- percona

--- a/k8s/dittofs-operator/pkg/percona/percona.go
+++ b/k8s/dittofs-operator/pkg/percona/percona.go
@@ -101,9 +101,9 @@ func BuildPerconaPGClusterSpec(ds *dittoiov1alpha1.DittoServer) (pgv2.PerconaPGC
 		spec.InstanceSets[0].DataVolumeClaimSpec.StorageClassName = cfg.StorageClassName
 	}
 
-	// Configure backups: explicitly disable unless configured.
+	// Configure backups: explicitly disable unless backups are configured and enabled.
 	// Percona defaults backups to enabled (IsEnabled returns true when Enabled is nil),
-	// so we must set Enabled=false when no backup repos are configured.
+	// so we must set Enabled=false when backups are not configured or are disabled.
 	if cfg.Backup != nil && cfg.Backup.Enabled {
 		spec.Backups = buildBackupsSpec(ds.Name, cfg.Backup)
 	} else {

